### PR TITLE
hv: timer modularization

### PR DESCRIPTION
**Pull request by ylian16**
This patch set provides modularization for timer. Tracked on #5920.

### CODE REVIEW

**Review by dongyaozu — APPROVED** [review]

**Review by fitchbe — APPROVED** [review]

**Comment by wenlingz** [comment]
@ylian16 please rebase your code, thanks

**Comment by wenlingz** [comment]
@ylian16 please check your code, fail to launch waag(waag hang) in TGL hybrid-rt and WHL industry and EHL hybrid-rt. 
please refer log:

root@1921681219sos-dom0:~/waag# cd /home/root/waag && bash launch_uos_id1.sh
tap device existed, reuse tap_WaaG
logger: name=console, level=4
logger: name=kmsg, level=3
logger: name=disk, level=5
SW_LOAD: get ovmf path /usr/share/acrn/bios/OVMF.fd, size 0x200000
vm_create: post_vm_id1
VHM api version 1.0
vm_setup_memory: size=0x100000000
open hugetlbfs file /run/hugepage/acrn/huge_lv1/post_vm_id1/D279543825D611E8864ECB7A18B34643
open hugetlbfs file /run/hugepage/acrn/huge_lv2/post_vm_id1/D279543825D611E8864ECB7A18B34643
level 0 free/need pages:1/1 page size:0x200000
level 1 free/need pages:11/4 page size:0x40000000

try to setup hugepage with:
	level 0 - lowmem 0x0, biosmem 0x200000, highmem 0x0
	level 1 - lowmem 0x80000000, biosmem 0x0, highmem 0x80000000
total_size 0x1c0000000

mmap ptr 0x0x7f1b85ea8000 -> baseaddr 0x0x7f1bc0000000
mmap 0x80000000@0x7f1bc0000000
touch 2 pages with pagesz 0x40000000
mmap 0x80000000@0x7f1cc0000000
touch 2 pages with pagesz 0x40000000
mmap 0x200000@0x7f1cbfe00000
touch 1 pages with pagesz 0x200000

really setup hugepage with:
	level 0 - lowmem 0x0, biosmem 0x200000, highmem 0x0
	level 1 - lowmem 0x80000000, biosmem 0x0, highmem 0x80000000
vm_init_vdevs
No pm notify channel given
pci init hostbridge

pci init lpc

pci init passthru

polling 37...
Listening 37...
pci init virtio-blk

pci init virtio-net

pci init igd-lpc

tpm: init_vtpm2:Invalid socket path!
acrn_sw_load
SW_LOAD: partition blob /usr/share/acrn/bios/OVMF.fd size 0x200000 copied to addr 0x7f1cbfe00000
SW_LOAD: build e820 6 entries to addr: 0x7f1bc00ef008

SW_LOAD: entry[0]: addr 0x0000000000000000, size 0x00000000000a0000,  type 0x1

SW_LOAD: entry[1]: addr 0x0000000000100000, size 0x000000003b6fc000,  type 0x1

SW_LOAD: entry[2]: addr 0x000000003b7fc000, size 0x0000000004004000,  type 0x2

SW_LOAD: entry[3]: addr 0x000000003f800000, size 0x0000000040800000,  type 0x1

SW_LOAD: entry[4]: addr 0x00000000e0000000, size 0x0000000020000000,  type 0x2

SW_LOAD: entry[5]: addr 0x0000000100000000, size 0x0000000080000000,  type 0x1

SW_LOAD: ovmf_entry 0xfffffff0
add_cpu

File: hypervisor/include/arch/x86/asm/tsc.h [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+
+#ifndef ARCH_X86_TSC_H
+#define ARCH_X86_TSC_H
+
+#include <types.h>
+
+#define TSC_PER_MS	((uint64_t)get_tsc_khz())
+
+/**
+ * @brief read tsc.
```
Comment (gvancuts) [inline hypervisor/include/arch/x86/asm/tsc.h:15]: Can you be a bit more verbose here, i.e.: 
```suggestion
 * @brief Read Time Stamp Counter (TSC)
```

File: hypervisor/include/arch/x86/asm/tsc.h [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+ *
+ * @return tsc value
```
Comment (gvancuts) [inline hypervisor/include/arch/x86/asm/tsc.h:17]: Same comment here, could you be a bit more verbose, e.g.:
```suggestion
 * @return Returns the TSC value (in ticks)
```

File: hypervisor/include/arch/x86/asm/tsc.h [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+static inline uint64_t rdtsc(void)
+{
+	uint32_t lo, hi;
+
+	asm volatile("rdtsc" : "=a" (lo), "=d" (hi));
+	return ((uint64_t)hi << 32U) | lo;
+}
+
+/**
+ * @brief  Get tsc.
```
Comment (gvancuts) [inline hypervisor/include/arch/x86/asm/tsc.h:28]: Same comment, can you be more verbose. What is the difference between `read tsc` and `get tsc` for example? I kind of guess that it's mainly about the unit used for the return value, is that it? In any case, more verbose text for the `@brief` and `@return` strings should make it immediately clear what the difference is (and what the function is about).

File: hypervisor/include/common/delay.h [hunk 4]
```diff
@@ -0,0 +1,21 @@
+/*
+ * Copyright (C) 2021 Intel Corporation. All rights reserved.
```
Comment (gvancuts) [inline hypervisor/include/common/delay.h:2]:
```suggestion
 * Copyright (C) 2021 Intel Corporation.
```

File: hypervisor/include/arch/x86/asm/tsc.h [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+ *
+ * @return tsc(KHz)
+ */
+uint32_t get_tsc_khz(void);
+
+/**
+ * @brief Calibrate tsc.
```
Comment (gvancuts) [inline hypervisor/include/arch/x86/asm/tsc.h:35]: Same comment. The explanation here is exactly the name of the function.

File: hypervisor/include/common/ticks.h [hunk 6]
```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright (C) 2021 Intel Corporation. All rights reserved.
```
Comment (gvancuts) [inline hypervisor/include/common/ticks.h:2]:
```suggestion
 * Copyright (C) 2021 Intel Corporation.
```

File: hypervisor/include/hw/hw_timer.h [hunk 7]
```diff
@@ -0,0 +1,15 @@
+/*
+ * Copyright (C) 2018 Intel Corporation. All rights reserved.
```
Comment (gvancuts) [inline hypervisor/include/hw/hw_timer.h:2]: If new file:
```suggestion
 * Copyright (C) 2021 Intel Corporation.
```

**Review by gvancuts — CHANGES_REQUESTED** [review]
A general comment, I would like to see a bit more attention paid to the documentation for each function. I know some of the function names are kind of self-explanatory but if you land on a documentation page, read about the brief explanation of a function just to see that it's the name of the function itself, it does not add any value (and makes it look like the programmer did not care much). Let's try to do a better job there ;-)

**Comment by wenlingz** [comment]
@ylian16 usually code PR and doc PR are generated separately

**Review by dongyaozu — APPROVED** [review]

**Review by dongyaozu — APPROVED** [review]

**Pull request merged**